src/xcode/ENA/ENA/Source/Models/Metadata/Test Result Metadata/__tests__/TestResultMetadataTests.swift

```diff
@@ -47,69 +45,61 @@ class TestResultMetadataServiceTests: XCTestCase {
 			riskCalculationResult.numberOfDaysWithCurrentRiskLevel,
 			"incorrect days since recent riskLEvel"
 		)
-		
+
 		// the for low risk the value should always be -1
 		XCTAssertEqual(secureStore.testResultMetadata?.hoursSinceHighRiskWarningAtTestRegistration, -1, "incorrect hours")
 	}
-	
+
 	func testUpdatingTestResult_ValidResult_NotPreviousTestResultStored() {
 		let secureStore = MockTestStore()
 		let riskCalculationResult = mockRiskCalculationResult(risk: .low)
 		secureStore.riskCalculationResult = riskCalculationResult
-		let sut = TestResultMetadataService(store: secureStore)
 
 		if let registrationDate = Calendar.current.date(byAdding: .day, value: -4, to: Date()) {
-			sut.registerNewTestMetadata(date: registrationDate, token: "Token")
+			Analytics.collect(.testResultMetadata(.registerNewTestMetadata(registrationDate, "")))
 		} else {
 			XCTFail("registration date is nil")
 		}
 
-		sut.updateResult(testResult: .positive, token: "Token")
+		Analytics.collect(.testResultMetadata(.updateTestResult(.positive, "")))
 		XCTAssertEqual(secureStore.testResultMetadata?.testResult, TestResult.positive, "incorrect testResult")
 		XCTAssertEqual(secureStore.testResultMetadata?.hoursSinceTestRegistration, (24 * 4), "incorrect hoursSinceTestRegistration")
 	}
-	
+
 	func testUpdatingTestResult_ValidResult_previouslyStoredWithSameValue() {
 		let secureStore = MockTestStore()
 		let riskCalculationResult = mockRiskCalculationResult(risk: .low)
 		secureStore.riskCalculationResult = riskCalculationResult
-		let sut = TestResultMetadataService(store: secureStore)
+		Analytics.collect(.testResultMetadata(.updateTestResult(.positive, "")))
 
 		if let registrationDate = Calendar.current.date(byAdding: .day, value: -4, to: Date()) {
-			sut.registerNewTestMetadata(date: registrationDate, token: "Token")
-			sut.updateResult(testResult: .positive, token: "Token")
-			secureStore.testResultMetadata?.hoursSinceTestRegistration = 0
-
+			Analytics.collect(.testResultMetadata(.registerNewTestMetadata(registrationDate, "")))
 		} else {
 			XCTFail("registration date is nil")
 		}
 
-		sut.updateResult(testResult: .positive, token: "Token")
+		Analytics.collect(.testResultMetadata(.updateTestResult(.positive, "")))
 		XCTAssertEqual(secureStore.testResultMetadata?.testResult, TestResult.positive, "incorrect testResult")
-		
-		/* The date shouldn't be updated if the test result is the same as the old one
-			- hoursSinceTestRegistration if updated should be (24 * 4)
-			- we explicitly set it into 0 in line 81, so we can see the change
-		*/
-		XCTAssertNotEqual(secureStore.testResultMetadata?.hoursSinceTestRegistration, (24 * 4), "incorrect hoursSinceTestRegistration")
+
+		// The date shouldn't be updated if the test result is the same as the old one
+		XCTAssertEqual(secureStore.testResultMetadata?.hoursSinceTestRegistration, (24 * 4), "incorrect hoursSinceTestRegistration")
```

Comment:
This was previously `XCTAssertNotEqual` is this new check for equality intentional?

Reply:
Oh yes there you found a pretty one. I changed it not correctly, because there were missing a line. So the test did something wrong after my changes!